Add `regexp/no-useless-flag` rule (#139)
ota-meshi/eslint-plugin-regexp: docs/rules/no-useless-flag.md

---
pageClass: "rule-details"
sidebarDepth: 0
title: "regexp/no-useless-flag"
description: "disallow unnecessary regex flags"
---
# regexp/no-useless-flag

> disallow unnecessary regex flags

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## :book: Rule Details

This will point out present regex flags that do not change the pattern.

### `i` flag (ignoreCase)

The `i` flag is only necessary if the pattern contains any characters with case

> variations. If no such characters are part of the pattern, the flag is

Reviewer: ```suggestion
variations. If the pattern contains no such characters, the flag will be
```

unnecessary. E.g. `/\.{3}/i`

<eslint-code-block fix>

```js
/* eslint regexp/no-useless-flag: "error" */

/* ✓ GOOD */
var foo = /a|b/i;

/* ✗ BAD */
var foo = /\.{3}/i;
var foo = /\w+/i;
```

</eslint-code-block>

### `m` flag (multiline)

> The `m` flag changes the meaning of the `^` and `$` anchors, so if the pattern
> doesn't contain these anchors, it's unnecessary. E.g. `/foo|[^\r\n]*/m`

Reviewer: ```suggestion
The `m` flag changes the behavior of the `^` and `$` assertions. If the pattern
doesn't contain these anchors, the `m` flag will be unnecessary. E.g. `/foo|[^\r\n]*/m`
```

<eslint-code-block fix>

```js
/* eslint regexp/no-useless-flag: "error" */

/* ✓ GOOD */
var foo = /^foo$/m;

/* ✗ BAD */
var foo = /foo|[^\r\n]*/m;
var foo = /a|b/m;
```

</eslint-code-block>

### `s` flag (dotAll)

The `s` flag makes the dot (`.`) match all characters instead of the usually

> non-line-terminator characters, so if the pattern doesn't contain a dot
> character set, it will be unnecessary. E.g. `/[.:]/s`

Reviewer: ```suggestion
non-line-terminator characters. If the pattern doesn't contain a dot
character set, the `s` flag will be unnecessary. E.g. `/[.:]/s`
```
